circus-train-core/src/test/java/com/hotels/bdp/circustrain/core/data/DefaultDataManipulatorFactoryManagerTest.java

```diff
@@ -0,0 +1,176 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.core.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import static com.hotels.bdp.circustrain.core.data.DefaultDataManipulatorFactoryManager.DATA_MANIPULATOR_FACTORY_CLASS;
+import static com.hotels.bdp.circustrain.core.data.DefaultDataManipulatorFactoryManagerTest.DataManipulatorType.HDFS;
+import static com.hotels.bdp.circustrain.core.data.DefaultDataManipulatorFactoryManagerTest.DataManipulatorType.S3_MAPREDUCE;
+import static com.hotels.bdp.circustrain.core.data.DefaultDataManipulatorFactoryManagerTest.DataManipulatorType.S3_S3;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.hadoop.fs.Path;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.hotels.bdp.circustrain.api.data.DataManipulator;
+import com.hotels.bdp.circustrain.api.data.DataManipulatorFactory;
+import com.hotels.bdp.circustrain.api.data.DataManipulatorFactoryManager;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DefaultDataManipulatorFactoryManagerTest {
+
+  private DataManipulatorFactory s3s3DataManipulatorFactory = new TestDataManipulatorFactory(S3_S3);;
+  private DataManipulatorFactory s3MapReduceDataManipulatorFactory = new TestDataManipulatorFactory(S3_MAPREDUCE);;
+  private DataManipulatorFactory hdfsDataManipulatorFactory = new TestDataManipulatorFactory(HDFS);;
+
+  private DataManipulatorFactoryManager manager;
+  private DataManipulatorFactory clientFactory;
+  private Path sourceLocation;
+  private Path replicaLocation;
+  private final String s3Path = "s3://<path>";
+  private final String hdfsPath = "hdfs://<path>";
+  private final Map<String, Object> copierOptions = new HashMap<>();
+
+  @Before
+  public void setup() {
+    manager = new DefaultDataManipulatorFactoryManager(
+        Arrays.asList(s3s3DataManipulatorFactory, s3MapReduceDataManipulatorFactory, hdfsDataManipulatorFactory));
+    sourceLocation = new Path(s3Path);
+  }
+
+  @Test
+  public void awsClientReturnedForS3S3Copy() {
+    replicaLocation = new Path(s3Path);
+    clientFactory = manager.getClientFactory(sourceLocation, replicaLocation, copierOptions);
+
+    assertTrue(((TestDataManipulatorFactory) clientFactory).getType() == S3_S3);
```

Comment:
I'd prefer using `assertEquals` here, you get a better error message from JUnit if this fails rather than just "wasn't true".

Reply:
Looks like assertEquals is deprecated!

Reply:
Are you sure? We use it all over the show, what does it suggest you use instead in the deprecation warning?

Reply:
`The method assertEquals(Object, Object) from the type Assert is deprecated. Add @SuppressWarnings 'deprecation' to 's3ManipulatorReturnedForS3S3Copy()'`

Reply:
`assertEquals(S3_S3, ((TestDataManipulatorFactory) clientFactory).getType());` looks OK to me? I don't see a warning in Eclipse?

Reply:
hmm yeah that works for me too. It's when I do `Assert.assertEquals..` it has an issue. Weird. Will change them 👍🏻

Reply:
Which `Assert` class do you import when you do that? My guess is you're by mistake pulling in an old Junit Assert class where everything is deprecated.

Reply:
ahh yeah looks like it, the one with everything deprecated is inside `junit.framework` but the Assert class that works is in `org.junit`.